webapp/src/shared/components/Button/Button.tsx

```diff
@@ -0,0 +1,43 @@
+import { ReactNode } from 'react';
+import './Button.css';
+
+export enum ButtonTypes {
+  SUBMIT = 'submit',
+  WARNING = 'warning',
+}
+
+type Buttonprops = {
+  buttonType: ButtonTypes;
+  buttonIcon?: JSX.Element;
+  action?: () => void;
+  disabled?: boolean;
+  children: ReactNode;
+};
```

Comment:
```suggestion
type ButtonProps = {
  variant: ButtonVariant;
  icon?: JSX.Element;
  onClick?: () => void;
  disabled?: boolean;
  children: string;
};
```

suggestion: We could change the variable names to have consistent naming across the codebase.

Reply:
I agree, done.